### ISSUE TITLE
Recursive schemas

### ISSUE THREAD

**Issue by kovasap** [metosin/malli#687]
I'm attempting to come up with a schema for hiccup, partially based on the example in https://github.com/metosin/malli#parsing-values (which is not a schema?).  What I have so far is:

```
(def Hiccup [:cat
             :keyword
             [:? [:map-of keyword? any?]]
             [:* [:schema Hiccup]]])
```

But I am getting cryptic `:child-errors`: `:malli.core/child-error {:type :schema, :properties nil, :children nil, :min 1, :max 1}`.  

Is this supported?  How could I accomplish it?

**Comment by frenchy64** [comment]
There's a schema for hiccup a few lines down from the link you posted:

```
(def Hiccup
  [:schema {:registry {"hiccup" [:orn
                                 [:node [:catn
                                         [:name keyword?]
                                         [:props [:? [:map-of keyword? any?]]]
                                         [:children [:* [:schema [:ref "hiccup"]]]]]]
                                 [:primitive [:orn
                                              [:nil nil?]
                                              [:boolean boolean?]
                                              [:number number?]
                                              [:text string?]]]]}}
   "hiccup"])

(def parse-hiccup (m/parser Hiccup))
```

**Comment by kovasap** [comment]
I noticed that before I made my post, but it seemed like such a deviation from basic example schemas (e.g. `[:vector :int]`) that I thought it was specifically for parsing.  I think I need to take some more time to unpack what's going on here.  For instance, the named pairs to `:catn` surprised me, as well as the `:registry` piece.

Good to know that this is actually a schema that can be used for all common operations!  That's helpful for me to know.
